plonky2/src/util/serialization/mod.rs

```diff
@@ -784,7 +783,16 @@ pub trait Read {
             num_lookup_polys,
             num_lookup_selectors,
             luts,
-        })
+        };
+
+        for _ in 0..gates_len {
```

Comment:
nit: can we do this using a `for_each`?

Reply:
I don't think we could, can we? There would be a `?` inside the closure.